app/models/sources/fog.rb

```diff
@@ -151,7 +163,9 @@ def provision_proxy(site)
     # Parameters:
     #   site - What site to add the found servers to (if any)
     def save_server(server, *args)
-      add_proxy(server.public_ip_address, config['proxy_port'], *args)
+      unless recent_decommissions.include? server.name
```

Comment:
Shouldn't this check be happening in `#find_orphaned_servers!`, since that method reports the number of new servers found?